### PR TITLE
Bl vectorized map estimates

### PR DESCRIPTION
**Pull request by Runjing-Liu120**
The map estimate methods can now take in multiple images at once

### CODE REVIEW

**Comment by Runjing-Liu120** [comment]
the `.get_metrics` method in sleep now uses the vectorized functions, and the tests pass on my machine. Feel free to merge after a review

File: bliss/models/encoder.py [hunk 1]
```diff
@@ -543,22 +543,34 @@ def sample_encoder(self, image, n_samples):
...
+        # shape (all) = (1 x (batchsize x n_ptiles) x max_detections x param_dim)
+        pred = self._get_var_params_for_n_sources(
+            h, tile_n_sources.flatten().unsqueeze(0)
+        )
+
+        # now reshape
+        pred = {
+            key: param.view(batchsize, n_ptiles, param.shape[2], param.shape[3])
+            for key, param in pred.items()
+        }
```
Comment (ismael-mendoza) [inline bliss/models/encoder.py:573]: excellent, thanks for figuring this out!

File: bliss/sleep.py [hunk 2]
```diff
@@ -342,31 +342,33 @@ def get_metrics(self, batch):
...
+        est_n_gal = estimates["galaxy_bool"].view(batch_size, -1).sum(-1)
+        true_n_gal = estimates["galaxy_bool"].view(batch_size, -1).sum(-1)
+        galaxy_counts_acc = (est_n_gal == true_n_gal).float().mean()
+
+        # accuracy of locations
+        est_locs = estimates["locs"]
+        true_locs = true_params["locs"]
+
+        mses = []
+        for i in range(batch_size):
```
Comment (ismael-mendoza) [inline bliss/sleep.py:361]: I think this for loop is OK for the locations, later we can improve it if necessary.

**Review by ismael-mendoza — APPROVED** [review]
Looks good, I just want to make sure it passes in the GPU and then I'll merge

**Pull request merged**